File: internal/domain/entities/storage_drives_groups_authorization_ids.go
package entities

import (
	"github.com/gofrs/uuid/v5"
)

type StorageDrivesGroupsAuthorizationsIDs struct {
	StorageDrivesID                      []uuid.UUID `json:"storage_drives_id,omitempty"`
	DirectoryGroupsID                    []uuid.UUID `json:"directory_groups_id,omitempty"`
	CreationIamGroupAuthorizationsID     []uuid.UUID `json:"creation_iam_group_authorizations_id,omitempty"`
	DeactivationIamGroupAuthorizationsID []uuid.UUID `json:"deactivation_iam_group_authorizations_id,omitempty"`
}

type storageDrivesGroupsAuthorizationIDsRepository struct {
	RepositoryName string

	StorageDrivesID                      string
	DirectoryGroupsID                    string
	CreationIamGroupAuthorizationsID     string
	DeactivationIamGroupAuthorizationsID string
}

func StorageDrivesGroupsAuthorizationIDsRepository() storageDrivesGroupsAuthorizationIDsRepository {
	return storageDrivesGroupsAuthorizationIDsRepository{
		RepositoryName: "storage_drives_groups_authorization_ids",

		StorageDrivesID:                      "storage_drives_id",
		DirectoryGroupsID:                    "directory_groups_id",
		CreationIamGroupAuthorizationsID:     "creation_iam_group_authorizations_id",
		DeactivationIamGroupAuthorizationsID: "deactivation_iam_group_authorizations_id",
	}
}

File: web/src/domain/entities/storage_drives_groups_authorization_ids.ts
namespace _StorageDrivesGroupAuthorizationIDs {
	export enum FieldColumn {
		StorageDrivesID = 'storage_drives_id',
		DirectoryGroupsID = 'directory_groups_id',
		CreationIamGroupAuthorizationsID = 'creation_iam_group_authorizations_id',
		DeactivationIamGroupAuthorizationsID = 'deactivation_iam_group_authorizations_id'
	}

	export const RepositoryName = 'storage_drives_groups_authorization_ids'

	export interface Interface {
		storage_drives_id?: string[]
		directory_groups_id?: string[]
		creation_iam_group_authorizations_id?: string[]
		deactivation_iam_group_authorizations_id?: string[]
	}
}

export default _StorageDrivesGroupAuthorizationIDs
